language/dsl/src/__tests__/helpers/asset-library.tsx

```diff
@@ -369,3 +375,113 @@ Info.AdditionalInfo = createSlot<{
   wrapInAsset: true,
   CollectionComp,
 });
+
+export interface ChoiceAsset extends AssetType<"choice"> {
+  /** A text-like asset for the choice's label */
+  title?: AssetWrapper;
+
+  /** Asset container for a note. */
+  note?: AssetWrapper;
+
+  /** The location in the data-model to store the data */
+  binding?: Binding;
+
+  /** The options to select from */
+  items?: Array<ChoiceItemType>;
+
+  /** Optional additional data */
+  metaData?: {
+    /** Additional data to send along with beacons */
+    beacon?: string | Record<string, any>;
+  };
+}
+
+export interface ChoiceItemType {
+  /** The id associated with the choice item */
+  id: string;
+
+  /** A text-like asset for the choice's label */
+  label?: AssetWrapper;
+
+  /** The value of the input from the data-model */
+  value?: string | number | boolean | null;
+}
+
+export const Choice = (
+  props: Omit<AssetPropsWithChildren<ChoiceAsset>, "binding"> & {
+    /** The binding */
+    binding: BindingTemplateInstance;
+  }
+) => {
+  const { binding, children, ...rest } = props;
+  return (
+    <Asset type="choice" {...rest}>
+      <property name="binding">{binding.toValue()}</property>
+      {children}
+    </Asset>
+  );
+};
+
+Choice.Title = createSlot<{
+  /** Some thing not in the asset  */
```

Comment:
Do we leverage that property (as we do on the `Input.Label`, for example)?

Reply:
Could you clarify what you mean by "leverage that property"?

Reply:
You are extending all the Choice's Slots props with `customLabelProp`. We do that for `Input.Label` to later be able to do that:
```tsx
<Input.Label customLabelProp="custom label slot value">
```
But it doesn't seem it's needed to Choice.Note, Choice.Items etc.